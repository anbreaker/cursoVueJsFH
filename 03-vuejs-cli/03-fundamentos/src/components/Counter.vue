<template>
  <div>
    <h1>{{ title || customTitle }}</h1>

    <h4>{{ counter }} <sup>2</sup> = {{ squareCounter }}</h4>

    <div>
      <button @click="increase">+1</button>
      <button @click="decrease">-1</button>
    </div>
  </div>
</template>

<script>
export default {
  name: "Counter",

  props: {
    title: String,

    start: {
      type: Number,
      default: 10,
      // required: true
    }
  },
  data() {
    return {
      counter: this.start
    };
  },

  methods: {
    getSquareValue() {
      console.log("getSquareValue");
      return this.counter * this.counter;
    },

    increase() {
      this.counter++;
    },

    decrease() {
      this.counter--;
    }
  },

  computed: {
    squareCounter() {
      console.log("squareCounter");
      return this.counter * this.counter;
    },

    customTitle() {
      return this.title || "Custom Counter";
    }
  }
};
</script>

<style scoped>
button {
  background-color: #64bb87;
  border-radius: 5px;
  border: 1px solid azure;
  color: azure;
  cursor: pointer;
  margin: 0 5px;
  padding: 5px 15px;
  transition: 0.3s ease-in-out;
}

button:hover {
  background-color: #d4d227;
  transition: 0.3s ease-in-out;
}
</style>