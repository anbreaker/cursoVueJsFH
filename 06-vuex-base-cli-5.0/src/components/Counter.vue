<template>
  <h1>Vuex - Counter</h1>
  <h2>Direct Access: {{ $store.state.counter.count }}</h2>
  <h2>Count Computed: {{ countComputed }}</h2>

  <button @click="increment">+1</button>
  <button @click="incrementBy">+5</button>
  <button @click="incrementRandomValue" :disabled="isLoading">Random</button>

  <h1>mapState</h1>
  <h2>mapState: {{ count }}</h2>
  <h2>lastMutation: {{ lastMutation }}</h2>
  <h2>lastRandomValue: {{ lastRandomValue }}</h2>

  <h2>Direct Getter: {{ $store.getters['counter/squareCount'] }}</h2>
</template>

<script>
import { mapState, mapActions } from 'vuex'

export default {
  computed: {
    ...mapState('counter', [
      'count',
      'lastMutation',
      'lastRandomValue',
      'isLoading',
    ]),
    // ...mapState({
    //   count: (state) => state.count,
    // //  lastMutation: (state) => state.lastMutation,
    //   lastMutation: 'lastMutation',
    // }),

    countComputed() {
      return this.$store.state.counter.count
    },
  },
  methods: {
    increment() {
      this.$store.commit('counter/increment')
    },
    incrementBy() {
      this.$store.commit('counter/incrementBy', 5)
    },
    ...mapActions('counter', ['incrementRandomValue']),
  },
}
</script>

<style scoped>
button {
  background-color: #41b883;
  border-radius: 5px;
  border: none;
  color: aliceblue;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  margin-left: 1rem;
  padding: 10px;
}

button:hover {
  background-color: #2c9f67;
}

button:disabled {
  background-color: #34473f;
}
</style>
