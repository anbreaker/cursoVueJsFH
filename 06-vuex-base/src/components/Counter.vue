<template>
  <div>
    <h1>Counter - Vuex</h1>
    <h2>Direct Access: {{ $store.state.count }}</h2>
    <h2>Count Computed: {{ countComputed }}</h2>

    <button>+1</button>
    <button>+5</button>
    <button>Random</button>
    <hr />
    <h1>mapState - Vuex</h1>
    <h2>mapState: {{ count }}</h2>
    <h2>mapState - LastMutation: {{ lastMutation }}</h2>
  </div>
</template>

<script>
import { mapState } from "vuex";

export default {
  computed: {
    countComputed() {
      return this.$store.state.count;
    },

    // ...mapState(["count"])
    ...mapState({
      count: state => state.count,
      lastMutation: state => state.lastMutation
    })
  }
};
</script>

<style scoped>
button {
  background-color: #41b883;
  color: azure;
  font-size: 1rem;
  font-weight: bold;
  padding: 10px;
}
</style>
