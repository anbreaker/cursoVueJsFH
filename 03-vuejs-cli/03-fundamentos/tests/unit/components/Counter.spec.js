import { shallowMount, mount } from '@vue/test-utils';

import Counter from '@/components/Counter';

describe('Counter Component', () => {
  // test('should be match with the snapshot', () => {
  //   const wrapper = shallowMount(Counter);

  //   expect(wrapper.html()).toMatchSnapshot();
  // });

  test('should have the default value "Counter"', () => {
    const wrapper = shallowMount(Counter);

    expect(wrapper.find('h2').exists()).toBe(true);

    const h2Value = wrapper.find('h2').text();

    expect(h2Value).toBe('Custom Counter');
  });
});
