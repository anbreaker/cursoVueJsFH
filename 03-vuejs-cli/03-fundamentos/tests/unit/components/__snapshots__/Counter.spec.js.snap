// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component should be match with the snapshot 1`] = `
<div>
  <h2>Custom Counter</h2>
  <p>
    100
    <sup>2</sup>
    = 10000
  </p>
  <div><button>+1</button> <button>-1</button></div>
</div>
`;
